optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/constraintweight/descriptor/ConstraintConfigurationDescriptor.java

```diff
@@ -0,0 +1,163 @@
+    // ************************************************************************
+
+    public void processAnnotations(DescriptorPolicy descriptorPolicy, ScoreDefinition scoreDefinition) {
+        processPackAnnotation(descriptorPolicy);
+        ArrayList<Method> potentiallyOverwritingMethodList = new ArrayList<>();
+        // Iterate inherited members too (unlike for EntityDescriptor where each one is declared)
+        // to make sure each one is registered
+        for (Class<?> lineageClass : ConfigUtils.getAllAnnotatedLineageClasses(constraintConfigurationClass, ConstraintConfiguration.class)) {
+            List<Member> memberList = ConfigUtils.getDeclaredMembers(lineageClass);
+            for (Member member : memberList) {
+                if (member instanceof Method && potentiallyOverwritingMethodList.stream().anyMatch(
+                        m -> member.getName().equals(m.getName()) // Short cut to discard negatives faster
+                                && ReflectionHelper.isMethodOverwritten((Method) member, m.getDeclaringClass()))) {
+                    // Ignore member because it is an overwritten method
+                    continue;
+                }
+                processParameterAnnotation(descriptorPolicy, member, scoreDefinition);
+            }
+            potentiallyOverwritingMethodList.ensureCapacity(potentiallyOverwritingMethodList.size() + memberList.size());
+            memberList.stream().filter(member -> member instanceof Method)
+                    .forEach(member -> potentiallyOverwritingMethodList.add((Method) member));
+        }
+        if (constraintWeightDescriptorMap.isEmpty()) {
+            throw new IllegalStateException("The constraintConfigurationClass (" + constraintConfigurationClass
+                    + ") must have at least 1 member with a "
+                    + ConstraintWeight.class.getSimpleName() + " annotation.");
+        }
+    }
+
+    private void processPackAnnotation(DescriptorPolicy descriptorPolicy) {
+        ConstraintConfiguration packAnnotation = constraintConfigurationClass.getAnnotation(ConstraintConfiguration.class);
+        if (packAnnotation == null) {
+            throw new IllegalStateException("The constraintConfigurationClass (" + constraintConfigurationClass
+                    + ") has been specified as a " + ConstraintConfigurationProvider.class.getSimpleName()
+                    + " in the solution class (" + solutionDescriptor.getSolutionClass() + ")," +
+                    " but does not have a " + ConstraintConfiguration.class.getSimpleName() + " annotation.");
+        }
+        // If a @ConstraintConfiguration extends a @ConstraintConfiguration, their constraintPackage might differ.
+        // So the ConstraintWeightDescriptors parse this too.
+        constraintPackage = packAnnotation.constraintPackage();
+        if (constraintPackage.equals("")) {
+            constraintPackage = constraintConfigurationClass.getPackage().getName();
```

Comment:
The `constraintPacakage` field is only assigned to and never read. This might be a bug. If it's not a bug, the field is unused and should be removed.

Reply:
That's a bug indeed. the @ConstraintWeight should inherit it.

Reply:
Well, it turns out it's not a bug - the ConstraintWeightDescriptor goes through the annotations again, due to extends complexity, but indeed we don't need this field for now. Removing it

Reply:
@ge0ffrey probably another candidate for additional test coverage - default and custom package in ConstraintConfiguration with constraints written in Drools. Another scenario - overriding a constraintPackage on the ConstraintWeight.